src/App.tsx

```diff
@@ -15,48 +17,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_BY_ALPHABET = 'Sort alphabetically',
+  SORT_BY_LENGTH = 'Sort by length',
+}
+
+interface SortParams {
+  sortField: SortType;
+  reverse: boolean;
+}
+
+function getPrepareGoods(goods: string[], { sortField, reverse }: SortParams) {
+  let prepareGoods = [...goods];
+
+  if (sortField) {
+    prepareGoods = prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORT_BY_ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.SORT_BY_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    return prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortParams, setSortParams] = useState<SortParams>({
+    sortField: SortType.SORT_BY_ALPHABET,
+    reverse: false,
+  });
+
+  const resetSorting = () => {
+    if (sortParams.sortField || sortParams.reverse) {
+      setSortParams({
+        sortField: SortType.SORT_BY_ALPHABET,
+        reverse: false,
+      });
+    }
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => {
+            setSortParams({ sortField: SortType.SORT_BY_ALPHABET, reverse: false });
+          }}
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info',
+            { 'is-light': sortParams.sortField !== SortType.SORT_BY_ALPHABET })
+          }
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => {
+            setSortParams({ sortField: SortType.SORT_BY_LENGTH, reverse: false});
+          }}
```

Comment:
move to an own function. this function should accept `key` and `value` arguments, spread current params object and set a new one
Example:
```
  const handleSortClick = (key, value) => {
    setSortParams(prev => ({...prev, [key]: value}));
  }
```